services/property-svc/internal/property/api/grpc.go

```diff
@@ -21,16 +23,52 @@ func NewPropertyService(aggregateStore hwes.AggregateStore) *PropertyGrpcService
 func (s *PropertyGrpcService) CreateProperty(ctx context.Context, req *pb.CreatePropertyRequest) (*pb.CreatePropertyResponse, error) {
 	propertyID := uuid.New()
 
-	subjectID, err := uuid.Parse(req.GetSubjectId())
-	if err != nil {
-		return nil, err
+	var setID *uuid.UUID
+	if req.SetId != nil {
+		id, err := uuid.Parse(req.GetSetId())
+		if err != nil {
+			return nil, err
+		}
+		setID = &id
+	}
+
+	var none *bool
+	var selectData *models.SelectData
+
+	switch ftData := req.FieldTypeData.(type) {
+	case *pb.CreatePropertyRequest_None:
+		none = &ftData.None
+	case *pb.CreatePropertyRequest_SelectData_:
+		selectData = &models.SelectData{
+			AllowFreetext: ftData.SelectData.GetAllowFreetext(),
+			SelectOptions: hwutil.Map(ftData.SelectData.Options, func(option *pb.CreatePropertyRequest_SelectData_SelectOption) models.SelectOption {
+				var description string
+				if option.Description != nil {
+					description = *option.Description
+				}
+				return models.SelectOption{
+					//TODO: ID, will be added with #677,
+					Name:        option.Name,
+					Description: description,
+				}
+			}),
+		}
+	default:
+		defaultValue := true
+		none = &defaultValue
 	}
-	if err := commandsV1.NewCreatePropertyCommandHandler(s.as)(ctx, propertyID, subjectID, req.GetSubjectType(), req.GetFieldType(), req.GetName()); err != nil {
+
+	fieldTypeData := models.FieldTypeData{
```

Comment:
Why is there still a none value here? Just make `fieldTypeData` be of type `*FieldTypeData` and nil it by default, else set it in the switch/case block. We only needed the none because of protobuf oneof.

Reply:
Do you mean that `fieldTypeData` should be nil by default and not
```
fieldTypeData = {
    none = true
}
```
?

Not quite sure why we should do it this way?